app/models/time_profile.rb

```diff
@@ -37,31 +38,31 @@ def global?
     profile_type == "global"
   end
 
-  def ts_in_profile?(ts, default_tz = DEFAULT_TZ)
-    ts = Time.parse(ts) if ts.kind_of?(String)
-    self.ts_day_in_profile?(ts, default_tz) && self.ts_hour_in_profile?(ts, default_tz)
+  def ts_in_profile?(tstamp, default_tz = DEFAULT_TZ)
+    tstamp = Time.zone.parse(tstamp) if tstamp.kind_of?(String)
+    ts_day_in_profile?(tstamp, default_tz) && ts_hour_in_profile?(tstamp, default_tz)
   end
 
-  def ts_hour_in_profile?(ts, default_tz = DEFAULT_TZ)
-    ts = Time.parse(ts) if ts.kind_of?(String)
-    hours.include?(ts.in_time_zone(tz_or_default(default_tz)).hour)
+  def ts_hour_in_profile?(tstamp, default_tz = DEFAULT_TZ)
+    tstamp = Time.zone.parse(tstamp) if tstamp.kind_of?(String)
```

Comment:
I wonder if this line changes things. Was the previous line using ITV or local?

Reply:
It was just doing `Time.parse`. I'm assuming that's local (?).

Reply:
I removed the `zone` call just to be safe.